File: gherkin/php/src/Parser/ParserContext.php
<?php

declare(strict_types=1);

namespace Cucumber\Gherkin\Parser;

use Cucumber\Gherkin\ParserException;
use Cucumber\Gherkin\Token;
use Cucumber\Gherkin\TokenMatcher;

final class ParserContext
{
    /**
     * @param list<Token> $tokenQueue
     * @param list<ParserException> $errors
     */
    public function __construct(
        public readonly TokenScanner $tokenScanner,
        public readonly TokenMatcher $tokenMatcher,
        public array $tokenQueue,
        public array $errors,
    ) {
    }
}

File: gherkin/dart/lib/src/parser/ParserContext.dart
import 'dart:collection';

import 'package:gherkin/exceptions.dart';
import 'package:gherkin/language.dart';

import 'ITokenMatcher.dart';
import 'ITokenScanner.dart';

/// Internal use only.
class ParserContext
{
  final ITokenScanner tokenScanner;
  final ITokenMatcher tokenMatcher;
  final Queue<Token> tokenQueue;
  final List<ParserException> errors;

  ParserContext(this.tokenScanner, this.tokenMatcher, this.tokenQueue, this.errors);
}
